uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js

```diff
@@ -71,38 +71,60 @@ define([
     });
 
     describe('confirm()', function () {
+      var expectedDocumentsParameters, expectedTasksParameters;
+      var expectedCaseId = _.uniqueId();
+      var expectedContactId = _.uniqueId();
+
       beforeEach(function () {
         spyOn($q, 'all').and.returnValue({ then: _.noop });
         spyOn(taskService, 'saveMultiple');
         spyOn(documentService, 'saveMultiple');
         spyOn(assignmentService, 'save').and.returnValue({
           then: function (callback) {
             // fakes the db permanence, attaching an id to the assignment on the scope
-            callback(_.assign(_.clone(scope.assignment), { id: _.uniqueId() }));
+            callback(_.assign(_.clone(scope.assignment), { id: expectedCaseId }));
           }
         });
       });
+
       beforeEach(function () {
         prepAssignment();
-        scope.confirm() && scope.$digest();
-      });
 
-      it('does not pass the original documents objects to the documentService', function () {
-        var documents = documentService.saveMultiple.calls.mostRecent().args[0];
+        expectedDocumentsParameters = getActivitiesToBeSentToAPI(scope.documentList);
+        expectedTasksParameters = getActivitiesToBeSentToAPI(scope.taskList);
 
-        expect(documents.every(function (doc, index) {
-          return doc !== scope.taskList[index];
-        })).toBe(true);
+        scope.confirm() && scope.$digest();
```

Comment:
To me this looks like `scope.confirm()` can return `false` and in that case no digest needed. I think I can see the reason you did that - to save 1 line of code. But since we use such syntax for conditions - better to use 2 lines to avoid confusion.

Reply:
Thanks, it was like that, I should have fixed it. I removed the `scope.$digest()` statement because it's not needed.